sdks/python/apache_beam/runners/runner.py

```diff
@@ -116,8 +116,21 @@ class PipelineRunner(object):
   materialized values in order to reduce footprint.
   """
 
-  def run(self, pipeline):
-    """Execute the entire pipeline or the sub-DAG reachable from a node."""
+  def run(self, transform, options=None):
+    """Run the given transform with this runner.
+    """
+    # Imported here to avoid circular dependencies.
+    # pylint: disable=wrong-import-order, wrong-import-position
+    from apache_beam.pipeline import Pipeline
+    p = Pipeline(runner=self, options=options)
+    p | transform
+    return p.run()
```

Comment:
In order to execute the transform eagerly, don't we have to add wait_until_finish?

result = p.run()
result.wait_until_finish()
return result

Reply:
This returns the result to the caller, which can wait (or not) as desired.